lib/utils/Validator.js

```diff
@@ -3,12 +3,46 @@ const ArgumentsValidationError = require('../errors/ArgumentsValidationError');
 
 class Validator {
   constructor(schema) {
-    this.validateArguments = new Ajv().compile(schema);
+    const ajv = new Ajv({
+      $data: true,
+    });
+
+    ajv.addKeyword('getTransactionsByAddressFromToRange50', {
+      schema: false,
+      validate(data) {
+        const { from } = data;
+        const { to } = data;
+        if (from !== 'undefined' && to !== 'undefined' && to - from > 50) {
+          return false;
+        }
+        return true;
+      },
+    });
+
+    ajv.addKeyword('getUTXOFromToRange1000', {
+      schema: false,
+      validate(data) {
+        const { from } = data;
+        const { to } = data;
+        if (from !== 'undefined' && to !== 'undefined' && (to - from > 1000)) {
+          return false;
+        }
+        return true;
+      },
+    });
+
+    this.validateArguments = ajv.compile(schema);
   }
 
   validate(args) {
     if (!this.validateArguments(args)) {
-      throw new ArgumentsValidationError(`params${this.validateArguments.errors[0].dataPath} ${this.validateArguments.errors[0].message}`);
+      if (this.validateArguments.errors[0].keyword === 'getTransactionsByAddressFromToRange50') {
+        throw new ArgumentsValidationError(`"from" (${args.from}) and "to" (${args.to}) range should be less than or equal to 50`);
```

Comment:
Why doing that at all? Validator supposed to be context agnostic

Reply:
what exactly to do?) insight->dashd throw exceptions when we pass params for 'from'/'to' that above ranges(50 for getTransactionsByAddress, 1000 for getUTXO) and dapi always returns internal error. so, we only limit this range at dapi level. the same we verify for min/max values for all params, addresses rules, etc.  Is this approach wrong?

Reply:
@abvgedeika Anton means that it's not the right place for this logic. I think you should move it to a command which raises this error.

Reply:
changed